### PR TITLE
Adding utility for flow on hosted service and spec to create cluster on hosted service

### PR DESCRIPTION
**Pull request by nshidlin**
Changes to cypress.json:
Added env variables:
"INTEGRATION_API_BASE_URL": "https://api-integration.6943.hive-integration.openshiftapps.com/", "STAGING_API_BASE_URL": "https://api.stage.openshift.com/",
"PRODUCTION_API_BASE_URL": "https://api.stage.openshift.com/",
"HOSTED_ENV": "",
"OCM_USER": "",
"OCM_USER_PASSWORD": ""

In shared.json:
*function to login through ocm portal
*function to logout through ocm portal
*function to make api call using jwt stored in cookies
*function to verify auto filled pull secret matches user pull secret
*function to create assisted cluster on hosted service

new spec to create a cluster in hosted service

### CODE REVIEW

File: cypress/integration/shared.js [hunk 1]
```diff
@@ -192,6 +198,112 @@ export const waitForClusterInstallation = () => {
...
+  //Login to ocm
+  cy.visit('');
+  cy.get('#username').should('be.visible');
+  cy.get('#username').type(userName);
+  cy.get('#username').should('have.value', userName);
+  cy.get('#login-show-step2').click();
+  cy.get('#password').should('be.visible');
+  cy.get('#password').type(password);
+  cy.get('#password').should('have.value', password);
+  cy.get('#kc-form-login').submit();
```
Comment (lalon4) [inline cypress/integration/shared.js:211]: how do we know that submit succeeded?

File: cypress/integration/shared.js [hunk 2]
```diff
@@ -192,6 +198,112 @@ export const waitForClusterInstallation = () => {
...
+  cy.get('#form-input-name-field').clear();
+  cy.get('#form-input-name-field').type(clusterName);
+  cy.get('form').submit();
+  cy.get('#button-download-discovery-iso').should('be.visible');
+  cy.get('#form-input-name-field').should('have.value', clusterName);
+};
+
+export const logOutOCM = () => {
+  cy.get('#UserMenu').as('userMenu');
+  cy.get('@userMenu').click();
```
Comment (ukalifon) [inline cypress/integration/shared.js:290]: No need for the alias - just do cy.get('#UserMenu').click()

File: cypress/integration/shared.js [hunk 3]
```diff
@@ -192,6 +198,112 @@ export const waitForClusterInstallation = () => {
...
+export const verifyClusterCreation = (clusterName) => {
+  // response handler for makeApiCall
+  const findClusterInList = (response) => {
+    const clusters = response.body;
+    const checkClusterName = (cluster) => clusterName.localeCompare(cluster.name) === 0;
+
+    cy.expect(clusters.some(checkClusterName)).to.be.true;
+  };
+
+  makeApiCall('api/assisted-install/v1/clusters', 'get', findClusterInList);
```
Comment (ukalifon) [inline cypress/integration/shared.js:304]: Try to use cy.request(). This is how I traverse the list of clusters from the API and find if mine is created:
    cy.request(API_ENDPOINT + '/api/assisted-install/v1/clusters').then((response) => {
      let i = 0;
      for (i = 0; i < response.body.length; i++) {
        if (response.body[i].name == 'test-cluster-1') {
          break;
        }
      }
      expect(i).to.be.lt(response.body.length);
    });

Comment (nshidlin) [reply cypress/integration/shared.js:304]: makeApiCall uses cy.request. On the hosted service requests need to be accompanied by a token; makeApiCall gets the token and attaches it to the request

File: cypress/fixtures/example.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (mareklibra) [inline cypress/fixtures/example.json:1]: Please do not commit this file, so far we do not use fixtures

File: cypress/integration-qe/memberUserCreateCluster.spec.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import {
+  createClusterHosted,
+  loginOCM,
+  verifyClusterCreation,
+  CLUSTER_NAME,
+  OCM_USER,
+  OCM_USER_PASSWORD,
+} from './shared';
+
+describe('Member User Can Created Cluster in hosted env', () => {
```
Comment (mareklibra) [inline cypress/integration-qe/memberUserCreateCluster.spec.js:10]: Type: `Created`
I think we do not need to use capital letters here

File: cypress/integration/shared.js [hunk 6]
```diff
@@ -192,6 +198,112 @@ export const waitForClusterInstallation = () => {
   cy.contains('div.pf-c-progress__description', 'Installed', { timeout: CLUSTER_CREATION_TIMEOUT });
 };
 
+export const loginOCM = (userName, password) => {
```
Comment (mareklibra) [inline cypress/integration/shared.js:201]: This file is becoming mess. Can we separate OCM-only things to a file i.e. 'ocm.js'?

File: cypress/integration/shared.js [hunk 7]
```diff
@@ -192,6 +198,112 @@ export const waitForClusterInstallation = () => {
...
+          env: 'integration',
+        },
+      });
+      break;
+    case 'production':
+      break;
+  }
+};
+
+export const makeApiCall = (apiPostfix, method, responseHandler, requestBody = {}) => {
```
Comment (lalon4) [inline cypress/integration/shared.js:230]: this makeApiCall is relevant only to OCM flow, so its name need to represent it somehow, i.e ocmMakeApiCall.

File: cypress/integration/ocmShared.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+// verifies cluster was created and associated to user
+export const verifyClusterCreationHosted = (clusterName) => {
+  // response handler for makeApiCall
+  const findClusterInList = (response) => {
+    const clusters = response.body;
+    const checkClusterName = (cluster) => clusterName.localeCompare(cluster.name) === 0;
+
+    cy.expect(clusters.some(checkClusterName)).to.be.true;
```
Comment (ukalifon) [inline cypress/integration/ocmShared.js:90]: should be enough to just write "expect", without "cy."

**Pull request closed without merging**